src/app/core/services/notification.service.ts

```diff
@@ -139,28 +136,33 @@ export class NotificationService {
             fcmNotifications.forEach(this.sendFCMNotification)
           }
           this.storage.set(StorageKeys.LAST_NOTIFICATION_UPDATE, Date.now())
-        }
+        } else Promise.reject()
```

Comment:
Sorry i know this has been present for a long time but maybe we can update it as this whole function can be simplified. 

1. Move the format notification and any other appropriate code to their respective conditional blocks. If we choose local notifications we don't want to format FCM notifications and vice-versa.
2. The workflow can be simplified by using fluent apis (filter, map, etc)
3. Also move the code under `DefaultNotificationType === 'LOCAL'` to a separate function like it is for FCM.

Reply:
Another thing we can do is create an interface (with functions like schedule, cancel, sendTest, etc) for creating notification service types and then include 2 implementations (like FcmNotificationService and LocalNotificaitonService) and then Inject them in the class conditional on the configured value (`FCM` or `LOCAL`)
Then we just call the function of the injected service without worrying about the actual implementation.
What do you think ?

Reply:
okay never mind if it's already done in #229

Reply:
Yup fixed in https://github.com/RADAR-base/RADAR-Questionnaire/pull/229/commits/49abc6a6bc350c131fb313135fe0930f6bc5b41e